ocpmodels/trainers/energy_trainer.py

```diff
@@ -435,14 +452,18 @@ def predict(self, loader, results_file=None, disable_tqdm=False):
             predictions["energy"].extend(out["energy"].tolist())
 
         if results_file is not None:
-            print(f"Writing results to {results_file}")
+            results_file_path = os.path.join(
+                self.config["cmd"]["results_dir"],
+                f"is2re_{results_file}_{rank}.npy",
+            )
+            print(f"Writing results to {results_file_path}")
 
             # EvalAI expects a list of dicts with ids and energies
             evalAI_results = []
             for sid, energy in zip(predictions["id"], predictions["energy"]):
                 evalAI_results.append({"id": sid, "energy": energy})
 
-            with open(results_file, "w") as resfile:
-                json.dump(evalAI_results, resfile)
+            with open(results_file_path, "wb") as resfile:
+                np.save(resfile, evalAI_results)
```

Comment:
Looks like you're saving just these as objects, which I believe simply pickles the objects. Did you benchmark to see how much faster this is compared to json?
If it's still slow, you could convert these into numpy arrays and save using `np.savez`

Reply:
Update - 

`
NPY
save(s): 12.07152453660965
load(s): 21.0965283036232
size: 917.654805 MB
`
`
json
save(s): 122.88421788215638
load(s): 41.24016536474228
size: 1754.49965 MB
`

`
pkl
save(s): 9.927891254425049
load(s): 20.672446691989897
size: 917.654486 MB
`
`
npz
save(s): 4.47386748790741
load(s): 0.0010351777076721192
size: 410.871863 MB
`

Will update this PR to save as NPZ. @abhshkdz @anuroopsriram

Reply:
Sounds good to me. How many samples are these space / time estimates for? A 25k split?

Reply:
1M images - 1 val S2EF split